generic/tests/edk2_stability_test.py

```diff
@@ -0,0 +1,43 @@
+import re
+
+from avocado.core import exceptions
+from virttest import error_context
+from virttest import utils_misc
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Qemu edk2 stability test:
+    1) Try to log into a guest
+    2) Check serial log information
+    3) Cycle the above process
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+    timeout = params.get_numeric("login_timeout", 240)
+    check_messgae = params.get("check_messgae")
+    vms = env.get_all_vms()
+
+    def boot_check(info):
+        """
+        boot info check
+        """
+        return re.search(info, get_serial_console_output(), re.S)
+
+    def get_serial_console_output():
+        """
+        get the output of serail console
+        """
+        output = vm.serial_console.get_output()
+        return output
+
+    for vm in vms:
+        error_context.context("Check serial log result", test.log.info)
+        if not utils_misc.wait_for(
+                lambda: boot_check(check_messgae), timeout, 5):
```

Comment:
I think using `session.read_until_last_line_matches` would be better, it is like tail, not need to read the entire content of the console output many times.

Reply:
After comparison, this method uses edk2_stability_test.arm64-pci: PASS (35.08 s)
The original method takes edk2_stability_test.arm64-pci: PASS (16.77 s)

So this method was not adopted